### PR TITLE
[SPARK-39326][PYTHON][PS] replace "NaN" with real  "None" value in indexes

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?

Since pandas 1.4 https://github.com/pandas-dev/pandas/commit/aaba0efd630ed607c5aaaef7b5f43d2fe90ca81c

> Series.__repr__() and DataFrame.__repr__() no longer replace all null-values in indexes with “NaN” but use their real string-representations. “NaN” is used only for float("nan")

We need to fix the doctest in PS

### Why are the changes needed?
follow pandas behavior

### Does this PR introduce _any_ user-facing change?
Yes, follow pandas behavior

### How was this patch tested?
CI passed

### CODE REVIEW

**Comment by Yikun** [comment]
We need to cleanup all these doctest after bump pandas to 1.4. will address together in SPARK-39150.

File: python/pyspark/pandas/frame.py [hunk 1]
```diff
@@ -551,11 +551,11 @@ def ndim(self) -> int:
         >>> df = ps.DataFrame([[1, 2], [4, 5], [7, 8]],
         ...                   index=['cobra', 'viper', None],
         ...                   columns=['max_speed', 'shield'])
-        >>> df
+        >>> df  # doctest: +SKIP
```
Comment (itholic) [inline python/pyspark/pandas/frame.py:554]: qq: Why do we need to skip this?

File: python/pyspark/pandas/frame.py [hunk 2]
```diff
@@ -7226,36 +7226,36 @@ def sort_index(
...
+                A
+        b     2.0
+        a     1.0
+        None  NaN
+
+        >>> df.sort_index(na_position='first')  # doctest: +SKIP
+                A
+        None  NaN
+        a     1.0
+        b     2.0
```
Comment (itholic) [inline python/pyspark/pandas/frame.py:7245]: ditto?

File: python/pyspark/pandas/frame.py (hunk 1 above)
Comment (Yikun) [reply python/pyspark/pandas/frame.py:554]: The fix will be failed with panda 1.3.x (it's [current version of ci infra](https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage/blob/ac09d7a4431b209caa86090c54bfccd4d744d0ce/Dockerfile#L32)), as I mentioned in [previous comment](https://github.com/apache/spark/pull/36712#issuecomment-1140234328) , we will cleanup `# doctest: +SKIP` after we upgrade pandas to 1.4.

Comment (Yikun) [reply python/pyspark/pandas/frame.py:554]: BTW, here is the final test with pandas 1.4. all changes have been submited as PRs

[1] https://github.com/Yikun/spark/pull/102

**Comment by Yikun** [comment]
also cc @HyukjinKwon @xinrong-databricks ready for review

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master

**Pull request closed without merging**